store/persistent/benchmarks.go

```diff
@@ -0,0 +1,134 @@
+package persistent
+
+import (
+	"context"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/storetheindex/importer"
+	"github.com/filecoin-project/storetheindex/store"
+	"github.com/ipfs/go-cid"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+)
+
+const testDataDir = "../../test_data/"
+const testDataExt = ".data"
+
+func prepare(s store.Storage, size string, b *testing.B) {
+	out := make(chan cid.Cid)
+	errOut := make(chan error, 1)
+
+	file, err := os.OpenFile(testDataDir+size+testDataExt, os.O_RDONLY, 0644)
+	if err != nil {
+		b.Fatalf("couldn't find the right input file for %v, try synthetizing from CLI: %v", size, err)
+	}
+	defer file.Close()
+
+	p, _ := peer.Decode("12D3KooWKRyzVWW6ChFjQjK4miCty85Niy48tpPV95XdKu1BcvMA")
+	imp := importer.NewCidListImporter(file)
+
+	go imp.Read(context.Background(), out, errOut)
+	for c := range out {
+		err = s.Put(c, p, c)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+	err = <-errOut
+	if err != nil {
+		b.Fatal(err)
+	}
+
+}
+
+func read(s store.Storage, size string, m *metrics, b *testing.B) {
+	out := make(chan cid.Cid)
+	errOut := make(chan error, 1)
+
+	file, err := os.OpenFile(testDataDir+size+testDataExt, os.O_RDONLY, 0644)
+	if err != nil {
+		b.Fatal(err)
+	}
+	defer file.Close()
+
+	imp := importer.NewCidListImporter(file)
+
+	b.ResetTimer()
+	go imp.Read(context.Background(), out, errOut)
+	for c := range out {
+		now := time.Now()
+		_, found, err := s.Get(c)
+		if err != nil || !found {
+			b.Errorf("cid not found")
+		}
+		m.getTime.add(time.Since(now).Microseconds())
+
+	}
+	err = <-errOut
```

Comment:
I really like this pattern, collect results in range loop and only check error after channel is closed.
👍